include/clap/ext/draft/background-operation.h

```diff
@@ -0,0 +1,92 @@
+#pragma once
+
+#include "../../plugin.h"
+
+static const char CLAP_EXT_BACKGROUND_OPERATION[] = "clap.background_operation.draft/0";
+
+/*
+
+This extension let the host execute some operations on a background thread.
+
+Some operations can be CPU intensive and require a considerable amount of time, and being force to
+run them on the main thread prevent parallelisation.
+
+For example, loading a state or activating a plugin can require a bit of processing.
+When loading a large project, with a thousand of plugin instances, it is desirable to be able
+to load the plugin state and activate the plugin in a background thread. This can significantly
+improve the loading time.
+
+Concurrency isn't a trivial thing, and this extension should only be implemented if the plugin has
+expensive tasks to perform during load state or activate.
+
+Here is how this extension works:
+
+1. Check which operation can be performed in background
+
+        plugin_bgop->is_supported(plugin, CLAP_EXT_STATE);
+
+2. Inform the plugin that the host is about to start a background operation
+
+        const bool can_start = plugin_bgop->about_to_start(plugin);
+
+   Once about_to_start() is called, no more call can be made on the main thread.
+   If the background thread couldn't be created then it is allowed to jump to the step 6. immediately.
+
+3. Inform the plugin that we've setup the background thread and are about to start (on the
+   background thread)
+
+        plugin_bgop->started(plugin);
+
+4. Perform the background operations
+
+        plugin_state->load(...)
```

Comment:
Must these calls be made on the [bgop-thread]?  My reading is yes, the thread must be the same, but I think this should be made explicit as the concept of "symbolic" threads exists elsewhere in the CLAP spec.

Reply:
What happens when the plug-in needs to call the host on the main thread as part of an operation here?  It's not allowed to main main-thread calls so some things may be difficult for the plug-in author to write

Reply:
This is good question, I'm not sure what's the best answer is:
1. allow callbacks on `bgop-thread`
2. delay the callbacks until `finished()`

My gut feeling is that 1. would be the right answer. I don't think it is realistic to have the plugin asking what interfaces it can call from the host on the background thread, so it'd have to be **ALL OF THEM**.